onnxruntime/core/providers/cpu/reduction/reduction_ops.h

```diff
@@ -8,12 +8,59 @@
 #include "core/common/optional.h"
 #include "core/framework/op_kernel.h"
 #include "core/providers/cpu/containers.h"
+#include "core/util/math.h"
 #include "core/util/math_cpuonly.h"
 #include "core/platform/threadpool.h"
+#include "core/common/safeint.h"
 #include <cmath>
 
 namespace onnxruntime {
 
+/* Common verficiations. */
+void OrtEnforce_ReduceAggregatorKR(const std::vector<int64_t>& fast_shape, const Tensor& output);
+void OrtEnforce_ReduceAggregatorRK(const std::vector<int64_t>& fast_shape, const Tensor& output);
+void OrtEnforce_ReduceAggregatorKRK(const std::vector<int64_t>& fast_shape, const Tensor& output);
+
+typedef enum class FastReduceKind : uint8_t {
+  kNone = 0,   // no fast implementation
+  kK = 1,      // kept dim = no reduce
+  kR = 2,      // reduced dim = all reduced
+  kKR = 4,     // kept dim, reduced dim
+  kRK = 8,     // reduced dim, kept dim
+  kKRK = 16,   // kept dim, reduced dim, kept dim
+  kEmpty = 32  // empty reduce
+} FastReduceKind;
+
```

Comment:
Sorry - forgot 'enum class' is painful if you want to use the values as bits. 

Let's just use 'enum FastReduceKind' for simplicity (also remove the 'typedef' and the duplicate type name at the end of the declaration).

Reply:
done